apps/trading/components/sidebar/sidebar.tsx

```diff
@@ -146,7 +163,9 @@ const SidebarButton = ({
         className={buttonClasses}
         data-testid={view}
         onClick={() => {
-          if (view === currView?.type) {
+          if (action) action();
```

Comment:
Not a huge fan of this. Rather than adding all the logic to detect 'zero content' can we just pass a generic onClick callback?

IMO something like this is better

```
const onSelect = (view: ViewType) => {
    if (view === currView?.type) {
      setView(null);
    } else {
      setView({ type: view });
    }
  };
  
  ...

<SidebarButton
  view={ViewType.ViewAs}
  icon={VegaIconNames.EYE}
  tooltip={t('View as')}
  onClick={() => {
    setViewAsDialogOpen(true);
  }}
/>
        <SidebarButton
          view={ViewType.Settings}
          icon={VegaIconNames.COG}
          tooltip={t('Settings')}
          onClick={() => onSelect(ViewType.Settings)}
        />
```